Added option to toggle convergence printout

sseraj opened the pull request:
## Purpose
<!--
Explain the goal of this PR, if it addresses an existing issue be sure to link to it.
Describe the big picture of your changes here, perhaps using a bullet list if multiple changes are done to accomplish a single goal.
If this PR accomplishes multiple goals, it may be best to create separate PR's for each.
-->
#27 uncommented the convergence history printout, which makes sense to have enabled by default. However, the output can be excessive when running large parameter sweeps. I added on option that can turn off the printout.

I also turned off the printout in the tests to make it easier to read the testflo output.

## Expected time until merged
<!--
Comment on whether or not this change is urgent and how long you expect this PR to take to review and be merged.
For example, a week would be a reasonable time frame for an average, non-urgent PR.
-->
2-3 days

## Type of change
<!--
What types of change is it?
Select the appropriate type(s) that describe this PR
-->

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [x] Maintenance update
- [ ] Other (please describe)

## Testing
<!-- Explain the steps needed to test the new code to verify that it does indeed address the issue and produce the expected behavior. -->
I ran cases with the `printConvergence` option set to True and False.

## Checklist
<!-- Put an `x` in the boxes that apply. -->

- [x] I have run `flake8` and `black` to make sure the Python code adheres to PEP-8 and is consistently formatted
- [ ] I have formatted the Fortran code with `fprettify` or C/C++ code with `clang-format` as applicable
- [x] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation

eytanadler commented on cmplxfoil/CMPLXFOIL.py:
```diff
@@ -267,6 +267,7 @@ def __call__(self, aeroProblem, useComplex=False, deriv=False):
         xfoil.cr09.minf1 = aeroProblem.mach  # Mach number
         xfoil.cr09.adeg = aeroProblem.alpha  # Angle of attack
         xfoil.ci04.itmax = self.getOption("maxIters")  # Iterations limit
+        xfoil.cl01.printconv = self.getOption("printConvergence")
```
Will this throw an error in complex mode since the variable exists only in the real Fortran source? If so, you could set the option within the `if`/`else` at the beginning of `__call__`.

sseraj replied:
This wasn't erroring out in complex mode, but setting the option in the if/else is definitely the right approach

eytanadler reviewed (commented):
Great idea, thanks for the fix. I added one comment/question about the behavior in complex mode.

Secondly, I originally tried to get complex printing to work too because it could help debug failing derivatives. However, printing the complex types was giving problems because it was the wrong data type. I managed to get a hacky solution working to debug my case, but I don't know enough about the Fortran printing incantations to get it working properly. If this would be an easy fix for you, it'd be great to have an option (set to `False` by default) to print complex mode convergence.

sseraj commented:
I added the complex convergence print out:
```
   6   rms: 1.2914E-06 -1.8613E-204i   max:  3.7569E-05 -5.7323E-203i    C at   80  2
       a =  2.00000 +1.0000E-200i   CL =  0.26478 +1.1292E-201i
      Cm = -0.01067 +4.0735E-205i   CD =  0.00521 +2.7093E-204i   =>   CDf =  0.00447 +7.9726E-205i   CDp =  0.00074 +1.9120E-204i

   7   rms: 7.5179E-09 -2.0075E-207i   max: -2.4694E-07 +6.6855E-206i    C at   80  2
       a =  2.00000 +1.0000E-200i   CL =  0.26478 +1.1292E-201i
      Cm = -0.01067 +4.0735E-205i   CD =  0.00521 +2.7093E-204i   =>   CDf =  0.00447 +7.9855E-205i   CDp =  0.00074 +1.9108E-204i

   8   rms: 8.7095E-12 -2.1738E-210i   max:  1.0561E-10 -1.4363E-208i    C at   80  2
       a =  2.00000 +1.0000E-200i   CL =  0.26478 +1.1292E-201i
      Cm = -0.01067 +4.0735E-205i   CD =  0.00521 +2.7093E-204i   =>   CDf =  0.00447 +7.9855E-205i   CDp =  0.00074 +1.9108E-204i
```
\
\
I also changed the real printout slightly.

Real convergence before:
```
   1   rms: 0.4412E-06   max: 0.1005E-04   C at   52  1
       a =  2.000      CL =  0.2648
      Cm = -0.0107     CD =  0.00521   =>   CDf =  0.00447    CDp =  0.00074

   2   rms: 0.4053E-08   max: -.1300E-06   C at   80  2
       a =  2.000      CL =  0.2648
      Cm = -0.0107     CD =  0.00521   =>   CDf =  0.00447    CDp =  0.00074

   3   rms: 0.1963E-10   max: 0.6209E-09   C at   80  2
       a =  2.000      CL =  0.2648
      Cm = -0.0107     CD =  0.00521   =>   CDf =  0.00447    CDp =  0.00074
```
\
Real convergence after:
```
   1   rms: 4.4117E-07   max:  1.0047E-05   C at   52  1
       a =  2.00000      CL =  0.26478
      Cm = -0.01067      CD =  0.00521   =>   CDf =  0.00447    CDp =  0.00074

   2   rms: 4.0530E-09   max: -1.3004E-07   C at   80  2
       a =  2.00000      CL =  0.26478
      Cm = -0.01067      CD =  0.00521   =>   CDf =  0.00447    CDp =  0.00074

   3   rms: 1.9628E-11   max:  6.2090E-10   C at   80  2
       a =  2.00000      CL =  0.26478
      Cm = -0.01067      CD =  0.00521   =>   CDf =  0.00447    CDp =  0.00074
```

eytanadler commented:
Thank you! That printout is a work of art and you’re Vincent van Seraj. Is this ready for review?

sseraj commented:
Thanks! And yes, it's ready for review

A-CGray reviewed (approved):
LGTM
